Source/WebKit/UIProcess/glib/DisplayVBlankMonitorDRM.cpp

```diff
@@ -216,63 +256,62 @@ std::unique_ptr<DisplayVBlankMonitor> DisplayVBlankMonitorDRM::create(PlatformDi
     }
 #endif
 
+    std::optional<DrmNodeWithCrtc> drmNodeWithCrtcInfo;
 #if PLATFORM(WPE) && ENABLE(WPE_PLATFORM)
-    String filename;
-    if (usingWPEPlatformAPI)
-        filename = String::fromUTF8(wpe_display_get_drm_device(wpe_display_get_primary()));
-    else
-        filename = WebCore::PlatformDisplay::sharedDisplay().drmDeviceFile();
-#else
-    auto filename = WebCore::PlatformDisplay::sharedDisplay().drmDeviceFile();
-#endif
-    if (filename.isEmpty()) {
-        RELEASE_LOG_FAULT(DisplayLink, "Could not create a vblank monitor for display %u: no DRM device found", displayID);
-        return nullptr;
-    }
+#ifdef WPE_PLATFORM_DRM
+    if (usingWPEPlatformAPI && WPE_IS_MONITOR_DRM(monitor)) {
+        String filename = String::fromUTF8(wpe_display_get_drm_device(wpe_display_get_primary()));
 
-    auto fd = UnixFileDescriptor { open(filename.utf8().data(), O_RDWR | O_CLOEXEC), UnixFileDescriptor::Adopt };
-    if (!fd) {
-        RELEASE_LOG_FAULT(DisplayLink, "Could not create a vblank monitor for display %u: failed to open %s", displayID, filename.utf8().data());
-        return nullptr;
+        if (filename.isEmpty()) {
+            RELEASE_LOG_FAULT(DisplayLink, "Could not create a vblank monitor for display %u: no DRM device found", displayID);
+            return nullptr;
+        }
+
+        UnixFileDescriptor fd = UnixFileDescriptor(open(filename.utf8().data(), O_RDWR | O_CLOEXEC), UnixFileDescriptor::Adopt);
+        if (!fd) {
+            RELEASE_LOG_FAULT(DisplayLink, "Could not create a vblank monitor for display %u: failed to open %s", displayID, filename.utf8().data());
+            return nullptr;
+        }
+
+        drmNodeWithCrtcInfo = DrmNodeWithCrtc { WTFMove(fd), { wpe_monitor_drm_get_crtc_index(WPE_MONITOR_DRM(monitor)), wpe_monitor_get_refresh_rate(monitor) } };
     }
+#endif
+#endif
 
 #if PLATFORM(GTK)
-    auto crtcInfo = findCrtc(fd.value(), monitor);
+    drmNodeWithCrtcInfo = findDrmNodeWithCrtc(monitor);
 #elif PLATFORM(WPE)
 #if ENABLE(WPE_PLATFORM)
-    std::optional<std::pair<uint32_t, uint32_t>> crtcInfo;
     if (usingWPEPlatformAPI) {
 #ifdef WPE_PLATFORM_DRM
-        if (WPE_IS_MONITOR_DRM(monitor))
-            crtcInfo = { wpe_monitor_drm_get_crtc_index(WPE_MONITOR_DRM(monitor)), wpe_monitor_get_refresh_rate(monitor) };
-        else
+        if (!WPE_IS_MONITOR_DRM(monitor))
 #endif
```

Comment:
So, at this point `drmNodeWithCrtcInfo` can only be initialized for DRM platform, so we can avoid this ifdef and DRM monitor check and simply do:

```
if (!drmNodeWithCrtcInfo)
    drmNodeWithCrtcInfo = findDrmNodeWithCrtc(monitor);
```